### ISSUE TITLE
Syncing in data points 

### ISSUE THREAD

**Issue by janagombitova** [akvo/akvo-flow-mobile#1116]
# Context

Alphonso captures data offline. He them comes to the office where he connects to the Internet and wants to sync the data out of Flow app to Flow backend. In the app it is as simple as connecting to the network, but he actually wants to take the explicit action. He wants to send the data so he knows the syncing is happening. 

# Problem 
After talking with Ilyasse today he said many times the enumerators come back home from the field and want to make sure the data gets synced out correctly. As Nikki says - they live for the green tick - so he wants to make Flow sync. 

What happens often now is that he then goes to the overflow menu and hits **sync** because he thinks this syncs data out of Flow, as the status of the green tick is synced. He does not know that in syncs data in and is then surprised. He also does not know he needs to go to Settings > Data and there hit on Send submissions.... He expects this action to be in the overflow menu as in the list of data points he see they are still only submitted and need to be synced. 

<img width="1280" alt="screen shot 2018-05-18 at 07 27 03" src="https://user-images.githubusercontent.com/12456965/40217413-e3dae992-5a6c-11e8-91f6-42702a57af6c.png">

# Opportunity
How might we let Alphonso sync data out of Flow - to re-trigger the action, from the data point list and not confuse him with syncing out and syncing in data? 

@Kiarii and @valllllll2000 What do you think?

**Comment by valllllll2000** [comment]
I think we need to find a way to have both sync in and sync out together with the datapoint list. Of course, it does not make sense to have it in settings so out of context. The question is how do we represent it???
I have been looking at other apps and they have several levels of navigation: a bottom bar and then something else at the top for example:
![screenshot_20180518-091225](https://user-images.githubusercontent.com/923280/40222934-837abe50-5a81-11e8-9e0e-c867670dd64c.png)
![screenshot_20180518-091128](https://user-images.githubusercontent.com/923280/40222935-839c7f9a-5a81-11e8-9626-4bff4c6385d9.png)
![screenshot_20180518-091122](https://user-images.githubusercontent.com/923280/40222936-83b6ef74-5a81-11e8-86c4-e54941f3b9fa.png)
So I also had this idea: why not have list and map tabs at the bottom. And at the top, in the list we can have 2 options: sync in and sync out (the rest of the menu stays the same).

**Comment by Kiarii** [comment]
It sounds like putting the sync in and out options next to each other would partly address the issue. The term sync might be misused (since by itself it means making sure what is in the server is exactly what is in the device) when used as "sync in" and "sync out". We might need to consider the use of other terms to be explicit about the underlying actions e.g. "Upload" submitted datapoints to Flow instance, or Download datapoints from Flow instance..

**Comment by stellanl** [comment]
I agree. To me _synchronize_ implies a two-way process, and _in_ and _out_ is not especially clear either.
I feel _upload_ and _download_ is much more universally understood as being respectively to and from the bigger, more central system. OTOH I'm a computer geek _and_ was born before Unix...

**Comment by janagombitova** [comment]
I am liking this discussion and absolutely agree: sync in and sync out is not clear. When speaking with Ilyasse about it, he said they explain it in French as 'send' and 'receive' if he directly translates it to English. So we indeed need to find an easy to understand word and make sure to also change the data point status, that is now 'synced' to reflect the new word

**Comment by Kiarii** [comment]
Could we go for "Upload" and "Download" and add these two to the survey contextual menu, this would imply a number of things:

- removing the "Send submitted data points to flow instance" from settings
- changing data point status "synced" to "uploaded"
- changing notifications corresponding to "Downloading" to reflect the new terminology e.g. for X data points downloaded and
- changing the survey's empty state screen, in place of "... or sync data points"  --> "... or download data points"

If there are other areas where this change might relate to please add them to the list

**Comment by janagombitova** [comment]
@valllllll2000 one small detail, for the Setting option: **Upload all submitted data points to Flow instance.**

**Comment by valllllll2000** [comment]
## Test plan
Make sure all the changes above were handled

**Comment by janagombitova** [comment]
@valllllll2000 I was thinking yesterday about the discussion we had around this issue about the need to have in the app Settings to option to upload all data points captured on the phone to Flow's backend. We agreed to keep that setting in place for now, but I am starting to have doubts about its value. Why? 
- many users do not even know it is there 
- when Aplhonso has two surveys he is capturing data for that day we uses the data point list as the place to check if all submission are uploaded correctly. So he selects one survey and checks the status of the data points, then selects the 2nd survey and checks the list of data points. When he sees that for the 1st one the points are not uploaded he will use the new item you created and then he would repeat that process again. 

I do understand that when uploading we should indeed handle all data points at once. So if Alphonso submits and Flow triggers the upload process automagically, Flow should check for everything that needs to be uploaded. 

But I was just thinking for Alphonso, I cannot really see a value in having the Setting available. What do you think?

**Comment by valllllll2000** [comment]
@janagombitova I think this should be handled in a separate issue https://github.com/akvo/akvo-flow-mobile/issues/1140

**Comment by kymni** [comment]
Text changes all in place. One question though, when the user taps on "Upload all submitted data points to Flow instance" they're taken back to the landing page with no other confirmation of whether the upload is happening or not. Tested a case where I disabled Internet and tapped on "Upload all submitted data points to Flow instance". App navigated back to landing page with no other action. Should we not notify the user that we're "Uploading" or "No connection" based on status?
